src/wallet/db.cpp

```diff
@@ -25,7 +25,13 @@ std::vector<fs::path> ListDatabases(const fs::path& wallet_dir)
         try {
             // Get wallet path relative to walletdir by removing walletdir from the wallet path.
             // This can be replaced by boost::filesystem::lexically_relative once boost is bumped to 1.60.
-            const fs::path path = it->path().string().substr(offset);
+            fs::path path = wallet_dir.string();
```

Comment:
You could still keep `path` const as before, by utilizing immediately-invoked lambda, as suggested in [ES.28: Use lambdas for complex initialization, especially of const variables](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-lambda-init) guideline, with something like this (also avoiding re-initializing `path` variable):

```
            const fs::path path = [&wallet_dir, &it, offset]() {

                if (wallet_dir == wallet_dir.root_path()) {
                   return it->path().string().substr(offset - 1);
                }

                return it->path().string().substr(offset);
            }();
```
Not sure if this is the best way to do that overall, as I don't have basically any experience with boost/stl filesystem...

Reply:
Will try this approach